awscli/customizations/s3/utils.py

```diff
@@ -519,6 +521,7 @@ def map_upload_part_params(cls, request_params, cli_params):
         """Map CLI params to UploadPart request params"""
         cls._set_sse_c_request_params(request_params, cli_params)
         cls._set_request_payer_param(request_params, cli_params)
+        cls._set_checksum_algorithm_param(request_params, cli_params)
```

Comment:
Something interesting I noticed is that the `map_upload_part_params` isn't actually called anywhere. Based on the [commit history](https://github.com/aws/aws-cli/commit/fc25229aeebf3f13721b307fd658a57d3c617246#diff-a06cd12835dfbf1c3436863e220d13fc24a1fab17ace4de80cce8fa4b9158409), I think this is actually dead code that wasn't cleaned up after integration with `s3transfer`.

Digging into `s3transfer`'s upload logic, it looks like `UploadSubmissionTask` handles both regular and [multipart uploads](https://github.com/aws/aws-cli/blob/9419d473fdb7f9017ff1c935da5b5cd8834ce86f/awscli/s3transfer/upload.py#L696). And the AWS CLI's `UploadRequestSubmitter` [always sets `map_put_object_params` as the request mapper](https://github.com/aws/aws-cli/blob/v2/awscli/customizations/s3/s3handler.py#L341).

You can verify this by commenting out this line and calling `aws s3 cp large-file s3://my-bucket --checksum-algorithm CRC32` and observing that `CreateMultipartUpload` and `UploadPart` are called with the checksum header.

For this PR, let's remove this line but keep the method for now. I want to avoid polluting this PR with changes not totally relevant to the task. We can create a backlog item to address potential dead code.